File: coffeetime-parent/coffeetime-scala/src/main/scala/org/mrseasons/coffeetime/scala/_11_class/Counter.scala
package org.mrseasons.coffeetime.scala._12_class

/**
 * Created by mrseasons on 2015/2/12.
 */
class Counter {
  private var value = 0

  //default public
  def increment(): Unit = {
    value += 1
  }

  def current = value
}

File: coffeetime-parent/coffeetime-kotlin/src/main/kotlin/org/mrseasons/coffeetime/kotlin/_11_class/Counter.kt
package org.mrseasons.coffeetime.kotlin._11_class

/**
 * Created by mrseasons on 2015/06/25.
 */
class Counter {
    private var value = 0
    var defaultValue = 1
    public var publicValue: Int = 2

    fun increment(): Unit {
        value += 1
    }

    fun current() = value
}